traffic_ops/testing/api/v4/deliveryservices_test.go

```diff
@@ -1238,6 +1254,272 @@ func VerifyPaginationSupportDS(t *testing.T) {
+	}
+}
+
+func GetDeliveryServiceByValidType(t *testing.T) {
+	if len(testData.DeliveryServices) > 0 {
+		firstDS := testData.DeliveryServices[0]
+
+		if firstDS.TypeID == nil && firstDS.Type != nil {
+			ty, _, err := TOSession.GetTypeByNameWithHdr(firstDS.Type.String(), nil)
+			if err != nil {
+				t.Fatalf("Error in Getting Type by Name: %v", err)
+			}
+			if len(ty) == 0 {
+				t.Fatalf("no Type named %v" + firstDS.Type.String())
+			}
+			firstDS.TypeID = &ty[0].ID
+			qparams := url.Values{}
+			qparams.Set("type", strconv.Itoa(*firstDS.TypeID))
+			resp, _, err := TOSession.GetDeliveryServicesV4(nil, qparams)
+			if err != nil {
+				t.Fatalf("Error in Getting Deliveryservice by Type:%v - %v", err, resp)
+			}
+			if len(resp) == 0 {
+				t.Errorf("No delivery service available for the Type %v", *firstDS.CDNName)
+			} else {
+				if *resp[0].Type != *firstDS.Type {
+					t.Fatalf("Type expected: %s, actual: %s", *firstDS.Type, *resp[0].Type)
+				}
+			}
+		}
+	}
+}
+
+func GetDeliveryServiceByInvalidType(t *testing.T) {
+	qparams := url.Values{}
+	qparams.Set("type", "10000")
+	resp, _, err := TOSession.GetDeliveryServicesV4(nil, qparams)
+	if err != nil {
+		t.Fatalf("Error!! Getting Deliveryservice by Invalid Type ID %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Fatalf("Error!! Invalid Type shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetDeliveryServiceByInvalidAccessibleTo(t *testing.T) {
+	qparams := url.Values{}
+	qparams.Set("accessibleTo", "10000")
+	resp, _, err := TOSession.GetDeliveryServicesV4(nil, qparams)
+	if err != nil {
+		t.Fatalf("Error!! Getting Deliveryservice by Invalid AccessibleTo %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Fatalf("Error!! Invalid AccessibleTo shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func GetDeliveryServiceByValidXmlId(t *testing.T) {
+	if len(testData.DeliveryServices) > 0 {
+		firstDS := testData.DeliveryServices[0]
+
+		resp, _, err := TOSession.GetDeliveryServiceByXMLIDNullableWithHdr(*firstDS.XMLID, nil)
+		if err != nil {
+			t.Fatalf("Error in Getting DeliveryServices by XML ID: %v - %v", err, resp)
+		}
+		if len(resp) == 0 {
+			t.Errorf("No delivery service available for the XML ID %v", *firstDS.XMLID)
+		} else {
+			if *resp[0].XMLID != *firstDS.XMLID {
+				t.Fatalf("Delivery Service Name expected: %s, actual: %s", *firstDS.XMLID, *resp[0].XMLID)
+			}
+		}
+	}
+}
+
+func GetDeliveryServiceByInvalidXmlId(t *testing.T) {
+	resp, _, err := TOSession.GetDeliveryServiceByXMLIDNullableWithHdr("test", nil)
+	if err != nil {
+		t.Fatalf("Error!! Getting Delivery service by Invalid ID %v", err)
+	}
+	if len(resp) >= 1 {
+		t.Fatalf("Error!! Invalid Xml Id shouldn't have any response %v Error %v", resp, err)
+	}
+}
+
+func SortTestDeliveryServices(t *testing.T) {
+	var header http.Header
+	var sortedList []string
+	resp, _, err := TOSession.GetDeliveryServicesV4(header, nil)
+	if err != nil {
+		t.Fatalf("Expected no error, but got %v", err.Error())
```

Comment:
nit but `.Error` is unnecessary when using the `%v` formatting parameter

Reply:
Removed .Error